pkg/proc/native/proc_linux.go

```diff
@@ -222,6 +222,7 @@ func (dbp *Process) trapWait(pid int) (*Thread, error) {
 			if err != nil {
 				if err == sys.ESRCH {
 					// thread died while we were adding it
+					delete(dbp.threads, int(wpid))
```

Comment:
I don't think we need this `delete` as we haven't added the new thread to the list yet.